src/main/object-uploader.js

```diff
@@ -207,25 +208,61 @@ export default class ObjectUploader extends Transform {
   }
 
   _flush(callback) {
-    // If it has been uploaded in a single packet, we don't have to do anything.
-    if (this.id === null) {
+    if (this.emptyStream) {
+      let method = 'PUT'
+      let headers = {
+        'Content-Length': 0,
+        'Content-Type': this.contentType
+      }
+      let options = {
+        method, headers,
+        query: '',
+        bucketName: this.bucketName,
+        objectName: this.objectName
+      }
+
+      this.client.makeRequest(options, '', 200, '', true, (err, response) => {
+        if (err) return callback(err)
+
+        let etag = response.headers.etag
+        if (etag) {
+          etag = etag.replace(/^"/, '').replace(/"$/, '')
+        }
+
+        // Ignore the 'data' event so that the stream closes. (nodejs stream requirement)
+        response.on('data', () => {})
+
+        // Give the etag back, we're done!
+        process.nextTick(() => {
+          this.callback(null, etag)
+        })
+
+        // Because we're sure the stream has ended, allow it to flush and end.
+        callback()
+      })
+
       return
     }
+    // If it has been uploaded in a single packet, we don't have to do anything.
+    else {
+      if (this.id === null) {
+        return
+      }
```

Comment:
you don't need else statement here @shruthied because you have return statement at the end of if block above.

i.e you can do it like :
```
if (this.emptyStream) {
    ....
    return
}
if (this.id == null) {
    return
}
```

Reply:
removed the else statement as mentioned @krishnasrinivas